checksit/generic.py

```diff
@@ -251,3 +252,67 @@ def check_var(dct, variable, defined_attrs, skip_spellcheck=False):
                     )
 
     return errors, warnings
+
+
+def check_file_name(file_name, vocab_checks=None, **kwargs):
+    """
+    Checks format of file name
+
+    Works for NCAS-GENERAL, would work for NCAS-RADAR if radar scan type is added as data product
+    """
+    vocab_checks = vocab_checks or {}
+    errors = []
+    warnings = []
+    file_name_parts = file_name.split("_")
+
+    # check instrument name
+    if "instrument" in vocab_checks.keys():
+        if vocabs.check(vocab_checks["instrument"], file_name_parts[0], label="_") != []:
+            errors.append(f"[file name]: Invalid file name format - unknown instrument {file_name_parts[0]}")
+    else:
+        msg = "No instrument vocab defined in specs"
+        raise KeyError(msg)
+
+    # check platform
+    if "platform" in vocab_checks.keys():
+        if vocabs.check(vocab_checks["platform"], file_name_parts[1], label="_") != []:
+            errors.append(f"[file name]: Invalid file name format - unknown platform {file_name_parts[1]}")
+    else:
+        msg = "No platform vocab defined in specs"
+        raise KeyError(msg)
+    
+    # check date format
+    # could be yyyy, yyyymm, yyyymmdd, yyyymmdd-HH, yyyymmdd-HHMM, yyyymmdd-HHMMSS
+    # first checks format, then date validity 
+    if not re.match("^\d{4}$|^\d{6}$|^\d{8}$|^\d{8}-\d{2}$|^\d{8}-\d{4}$|^\d{8}-\d{6}$", file_name_parts[2]):
```

Comment:
@joshua-hampton: this PR looks good. A minor point about using inline regular expressions. 

I would recommend creating the regex at the top of the module with `date_regex = re.compile("...")` and then using it with `date_regex.match(file_name_parts[2])`. This is generally more efficient and it means the complex regex is put somewhere else and could be easily tested in a unit test with a range of example files.